dora/core/client/fs/src/main/java/alluxio/client/file/ufs/UfsFileInStream.java

```diff
@@ -62,21 +65,28 @@ public int read() throws IOException {
       return -1;
     }
     mPosition++;
+    Metrics.BYTES_READ_UFS.inc(1);
     return res;
   }
 
   @Override
   public int read(ByteBuffer byteBuffer, int off, int len) throws IOException {
+    int totalBytesRead;
     if (byteBuffer.hasArray()) {
-      return read(byteBuffer.array(), off, len);
+      totalBytesRead = read(byteBuffer.array(), off, len);
+      if (totalBytesRead > 0) {
+        Metrics.BYTES_READ_UFS.inc(totalBytesRead);
```

Comment:
Instead of adding the metric accounting logic in one specific class, I'd recommend making this logic a dedicated wrapper, e.g. `MetricAccountingInputStream`, which takes in an underlying generic `InputStream`, and the metric key that the number of bytes read from the underlying stream would be attributed to. The `read` methods then delegate to the underlying stream, and adjust the metric according to the bytes read from it.

This approach has the advantage of
1. the metric accounting logic is separated from the main IO path, reducing the noise in code.
2. the metric accounting logic is pluggable and composable. You can use any combination of input stream and metric key. The metric key will not be tied to a particular class like `UfsFileInStream`. If some one creates a new stream `WorkerFirstThenFallbackToUfsInputStream` and forgets about plugging in the metric accounting, then you lose track of the bytes read from UFS there.

Reply:
`UfsFileInStream` is the only top-level FileInStream that is created and used in UfsBaseFileSystem.
Various UFS will be opened in the `UfsFileInStream`. 

I agree with you that your suggestion is a good idea. But let's do it later.